lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -378,6 +377,8 @@ ImageDecoderImpeller::UnsafeUploadTextureToPrivate(
   }
   blit_pass->EncodeCommands(context->GetResourceAllocator());
 
+  bool did_add_fence = context->AddTrackingFence(result_texture);
```

Comment:
Create the fence after the call to `CommandQueue::Submit`.

`glFenceSync` should be called after issuing all of the GL commands that write to the texture.  In theory that may not happen until after `Submit` returns successfully.

(In practice the `ReactorGLES::AddOperation` calls done during encoding will probably be able to call `React`, which will call the GL APIs)

Reply:
Done